src/drivers/storage/BundlrStorageDriver.ts

```diff
@@ -126,8 +126,7 @@ export class BundlrStorageDriver extends StorageDriver {
     const bundlr =
       this.metaplex.identity() instanceof WalletAdapterIdentityDriver
         ? new WebBundlr(address, currency, this.metaplex.identity(), options)
-        : new NodeBundlr(address, currency, this.metaplex.identity(), options);
-
+        : new NodeBundlr(address, currency, this.metaplex.identity().keypair.secretKey, options);
```

Comment:
If you want the type to work here on the KeypairIdentityDriver, you simply need to revert the ternary condition like so:

```ts
const identity = this.metaplex.identity();

identity instanceof KeypairIdentityDriver
        ? new NodeBundlr(address, currency, identity.keypair.secretKey, options)
        : new WebBundlr(address, currency, identity, options);
```

Reply:
This would work so long as the identityDriver object in Metaplex.js is of type 'any'. Since it was set to IdentityDriver, it could never be a KeypairIdentityDriver, and thus, never have access to the keypair.secretKey property.

Reply:
KeypairIdentityDriver extends IdentityDriver

Reply:
Right - ah! I see now. Ok, I made the suggested change (reverted Metaplex.ts back to original state and then changed the bundlr logic to use the ternary function)

Reply:
Thanks!